toolchain/lower/handle.cpp

```diff
@@ -196,12 +192,17 @@ auto HandleNameReference(FunctionContext& context, SemIR::NodeId node_id,
                          SemIR::NameReference node) -> void {
   auto type_node_id =
       context.semantics_ir().GetTypeAllowBuiltinTypes(node.type_id);
-  if (type_node_id == SemIR::NodeId::BuiltinFunctionType ||
-      type_node_id == SemIR::NodeId::BuiltinNamespaceType) {
+  if (type_node_id == SemIR::NodeId::BuiltinNamespaceType) {
     return;
   }
-  // TODO: Handle name references to globals
-  context.SetLocal(node_id, context.GetLocal(node.value_id));
+
+  auto target = context.semantics_ir().GetNode(node.value_id);
+  if (auto function_decl = target.TryAs<SemIR::FunctionDeclaration>()) {
+    context.SetLocal(node_id, context.GetFunction(function_decl->function_id));
+  } else {
```

Comment:
What about
```
  auto target = context.semantics_ir().GetNode(node.value_id);
  if (target.Is<SemIR::NamespaceDeclaration>()) {
    return;
  } else if (auto function_decl = target.TryAs<SemIR::FunctionDeclaration>()) {
    context.SetLocal(node_id, context.GetFunction(function_decl->function_id));
  } else {
```
Would that work instead of the `type_node_id` business?

Reply:
I think that'll work for now but will stop working when we add aliases -- we could have a name reference to an alias to a name reference to a namespace.

Reply:
Is that going to be a problem for aliases to functions as well?

Reply:
I don't think there's a problem for aliases to functions: an alias to a function should propagate the value produced by the `NameReference` that directly names the function, and a `NameReference` to the alias should likewise propagate the value.

(I think there is a problem for `NameReference`s to global aliases, in general, because we don't handle references to globals properly in lowering. But that's not specific to functions or namespaces.)